rfc/2023/2023-05-01-va-rfc-frontend-performance.md

```diff
@@ -0,0 +1,156 @@
+- Teams may need help improving performance. We can adopt a "we're all in this together" approach and empower teams to help each other
+- This could be perceived as a Platform responsibility. If VFS teams are empowered to make or propose improvements, this work can and should be shared across VFS and Platform teams.
+- People may question the data informing this proposal. The VA-specific research mentioned is from 2019, and broadband adoption has likely changed since then. The guidance provided by "The Performance Inequality Gap" is based on global measures, not US or Veteran-specific measures. We could spend time gathering more data on the performance experiences of VA.gov users.
+
+## Alternatives
+
+> List the alternative approaches
+> * There are always alternatives. What other alternative solutions were considered? Not considered?
+> * What are their strengths, weaknesses, risks? Why weren’t they chosen?
+> * Do not allow bias of a solution to show in this section, ensure each alternative has been considering seriously or do not list it in this section
+
+One alternative would be to adopt an hypermedia-centric approach and have `vets-api` generate HTML documents and "partials" in coordination with a lightweight hypertext-centric framework such as [htmx](https://htmx.org), a ~14KiB JavaScript library. This would shift a lot of logic to the server and minimize the amount of JavaScript, the major contributor to the performance issues facing VA.gov. This would represent a paradigm shift in how VA.gov applications are built, and require a significant investment in time and resources to adopt across all applications. This shift could be done incrementally, but still represents a major architectural change.
+
+Another alternative to adopting performance budgets at this time is to do more research on the issue of performance before adopting any corrective measures. We'd want to gather and analyze metrics and talk to Veterans about what their experience with VA.gov applications is. We certainly should gather and analyze information regardless of any improvements, but it seems reasonable to predict that some number of Veterans are experiencing poor performance based on the data at hand.
+
+## Diagrams
+
+Try to visually represent the proposal using a [diagram](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams). [Live editor](https://mermaid.live/)
+```mermaid
+graph TD
+    A[Public] -->|TIC| B(AWS)
+    B --> C{Reverse Proxy}
+    C --> D[vets-api]
+    C --> F[fa:fa-chess-rook Kong]
+```
+
+## Performance Scores
+
+The following table samples performance scores, as of 10 May 2023, from the [Frontend Performance Dashboard](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/frontend-support-dashboard/lighthouse-performance-report/):
```

Comment:
describe what a higher score vs a lower score means. which is better? what does it mean to be at 15%? Is that like a 10-second load time? What % gets a page to a reasonable 2s load time?